code-compass.el

```diff
@@ -567,33 +568,40 @@ code can infer it automatically."
     (when git-root
       (c/get-coupled-files-alist
        git-root
-       `(lambda (x)
+       (buffer-file-name)
+       `(lambda (x file-name)
           (message
            "Finished to update coupled files for %s and found %s coupled files."
            ,git-root
            (length x)))))))
 
+(defun c/show-coupled-files (files file-name)
+  (--> files
+    (--sort (> (string-to-number (nth 3 it)) (string-to-number (nth 3 other))) files) ;; sort by number of commits
+    (--sort (> (string-to-number (nth 2 it)) (string-to-number (nth 2 other))) files) ;; sort then by how often this file has changed
+    (-map (lambda (file)
+            (when (or (string= (file-truename file-name) (file-truename (car file)))
+                      (string= (file-truename file-name) (file-truename (nth 1 file))))
+              (car
+               (--remove (string= (file-truename file-name) (file-truename it)) (-take 2 file)))))
+          it)
+    (-remove 'null it)
+    (if (null it)
+        (message "No coupled file found!")
+      (let ((open-file (completing-read "Find coupled file: " it nil 't)))
+        (find-file open-file)
+        )))
+  )
+
 (defun c/find-coupled-files ()
   "Allow user to choose files coupled according to previous changes."
   (interactive)
-  (let ((choose-file
-         (lambda (files)
-           (--> files
-             (--sort (> (string-to-number (nth 3 it)) (string-to-number (nth 3 other))) files) ;; sort by number of commits
-             (--sort (> (string-to-number (nth 2 it)) (string-to-number (nth 2 other))) files) ;; sort then by how often this file has changed
-             (-map (lambda (file)
-                     (when (or (string= (file-truename (buffer-file-name)) (file-truename (car file)))
-                               (string= (file-truename (buffer-file-name)) (file-truename (nth 1 file))))
-                       (car
-                        (--remove (string= (file-truename (buffer-file-name)) (file-truename it)) (-take 2 file)))))
-                   it)
-             (-remove 'null it)
-             (if (null it)
-                 (message "No coupled file found!")
-               (completing-read "Find coupled file: " it nil 't))))))
-    (--> (vc-root-dir)
-      (c/get-coupled-files-alist it choose-file)
-      (when (f-file-p it) (find-file it)))))
+  (--> (vc-root-dir)
```

Comment:
could you rewrite this as a one liner?
`(c/get-coupled-files-alist (vc-root-dir) (buffer-file-name) (lambda (files file) (c/show-coupled-files files file)))`
The reason is that `vc-root-dir` will always be a directory or nil, so `(when (f-file-p root) (find-file root))` should always be skipped.